### DISCUSSION TITLE
How to build RDKit MinimalLib on Apple Silicon

### DISCUSSION THREAD

**Discussion by syedzayyan** [rdkit/rdkit#7275] [Development]
Hello!

I am trying to build RDKit MinimalLib on an M2 Macbook Pro and I was wondering if anyone had experience building it from scratch. I have tried following this guide to build the main codebase which works.

https://greglandrum.github.io/rdkit-blog/posts/2023-03-17-setting-up-a-cxx-dev-env2.html#part-3-setup-your-environment-and-then-run-the-tests

Trying to run and install emscripten is a whole issue. TIA.

**Comment by ptosco** [comment]
Hi, there is a `Dockerfile` available
https://github.com/rdkit/rdkit/blob/a8d4912f88ae2ea9ea7afa366ba5b9c0be09cb79/Code/MinimalLib/docker/Dockerfile
that will take away all the emscripten pain from you. Instructions how to use the `Dockerfile` are inside the file itself as comments.

**Reply by syedzayyan** [reply]
Hello. Thanks for the reply. Urrrm I am working with 8 gigs of RAM, and rebuilding kind of takes some (too much) of my resources. I was wondering if I could run it barebones on the Silicon.

**Reply by ptosco** [reply]
I think so, though I never tried, as I always do my builds on Linux.
The emsdk GitHub page reports compaitibility with Silicon Mac:
https://github.com/emscripten-core/emsdk
Installing emscripten is pretty simple these days - I always run emscripten barebones on my Linux box.
